### ISSUE TITLE
First slide of presentation mode doesn't get outline

### ISSUE THREAD

**Issue by AsaAyers** [bvaughn/react-presents#9]
First, I'm not completely sure if it's a bug in `react-presents` or `styled-components`.

This badge/outline is missing on first slide, but it will show up when you change slides.

![selection_053](https://cloud.githubusercontent.com/assets/324999/21036666/7917ce2e-bd7c-11e6-949a-6df7f7013b63.png)

My dev tools show that all of the components have the right data, but the `<ContentSlide` has the wrong class.

I have a Max Power solution that uses a `key` on `<ThemeProvider` to scrap the whole tree when `isPresenterMode` changes: https://github.com/AsaAyers/react-presents/commit/d82fdee0f3331200ff6f6a11408e2cdc3e89c9b7

> Homer: Kids, there's three ways to do things. The right way, the wrong way and the Max Power way!
> Bart: Isn't that the wrong way?
> Homer: Yeah, but faster!

I can also solve it by adding a key to `<ContentSlide`, but this has to be applied to every slide.

```js
const slide = ({ isPresenterMode }) => (
    <ContentSlide key={isPresenterMode}>
```

It seems like if you can replace `<ContentSlide` (both solutions do this) when `isPresenterMode` changes, it solves the problem. I think it may be related to timing somehow. `<PresenterModePlugin` sets `isPresenterMode` in the middle of `componentWillMount`, maybe `styled-components` doesn't like that.

**Comment by bvaughn** [comment]
I'm not sure I understand what you're reporting.

1. Go to the demo site, [here](https://bvaughn.github.io/react-presents/).
1. Type "p" to open a presenter tab
1. Check the presenter tab and verify that the "presenter" badge is shown
1. Using arrow keys (or buttons) navigate to the last slide then back to the first
1. Verify that the "presenter" badge is visible on each slide and looks the same

I think I'm going to need a repro case or something because from what I can see, everything looks like it's working fine.

PS Which version of React are you using? The demo site is built with 15.4.1. Are you using an older one by chance?

**Comment by AsaAyers** [comment]
I can't publish this to `github.io` because I messed up my subdomain, but the start of my project is at https://github.com/AsaAyers/silence-presentation.

I'm using v15.4.1, that seems to be what comes with `create-react-app` now.

**Comment by bvaughn** [comment]
Maybe you can just do a quick screen recording with Quicktime and post it so I can see? B'c I can't repro what you're describing.

PS Which browser + version?

**Comment by AsaAyers** [comment]
I'm using Chrome Version 55.0.2883.21 beta (64-bit) on Linux. Now you should be able to see this on https://asaayers.github.io/silence-presentation/ and I recorded a video

![ezgif com-optimize](https://cloud.githubusercontent.com/assets/324999/21057943/904e86fe-bdf0-11e6-9006-56b6cd7e6d38.gif)

**Comment by bvaughn** [comment]
I wonder what the difference is between https://asaayers.github.io/silence-presentation/ and https://bvaughn.github.io/react-presents/

Have you tried the same thing on https://bvaughn.github.io/react-presents/? (Can you confirm that it works fine there?)

**Comment by AsaAyers** [comment]
Thanks for your help with this. It does show up on first render for your presentation, but not mine. I've just copied and pasted examples with very few modifications. Here is my `App.js`, I'm just manually importing instead of doing the webpack trick to automatically load slides.

```js
import React from 'react';
import { Presentation, Slide, DropDownNav, PresenterModePlugin } from 'react-presents'

import Intro from './slides/intro'
import NotExpert from './slides/not-expert'
import Goals from './slides/goals'

const slides = [
    Intro,
    NotExpert,
    Goals,
]

const options = slides
  .map((slide, index) => ({
    label: slide.title,
    value: index
  }))
  .filter((option) => option.label)

export default () => (
    <Presentation>
        <PresenterModePlugin />
        {slides.map((Component, index) => (
            <Slide
                component={Component}
                key={index}
                />
        ))}
        <DropDownNav
            key='DropDownNav'
            options={options}
            />
    </Presentation>
)```

**Comment by bvaughn** [comment]
Hm... which version of `styled-components` are you using?

**Comment by AsaAyers** [comment]
```
asa@asa:~/repos/silence-presentation$ grep version node_modules/styled-components/package.json 
  "version": "1.1.2",
```
I'm not depending on it explicitly, it's whatever came down with `react-presents`.

When I downgrade to exactly 1.0.8 (based on your `package.json`) it works as expected. I'll see if I can isolate a minimal example and file a bug with them.

Thanks again for your help, I'll go ahead and close this since the bug isn't here.

Just in case you might be curious (I know I always am)

| styled-components | result |
| --------------------------| ------- |
| 1.0.8 | :+1:  |
| 1.0.9 | :+1: |
|1.0.10 | :red_circle:  |
|1.0.11 | :red_circle:  |
|1.1.2 | :red_circle:  |

**Comment by bvaughn** [comment]
That's interesting!

Lazy-ping @mxstbr. Any idea what may have changed between 1.0.9 and 1.0.10 that would cause the issue being reported here?

(If you're busy, no worries. I can dig into this later. I've been at work since it was reported and so I haven't been able to really take a look.)

**Comment by mxstbr** [comment]
I _think_ we fixed this again today in `v1.1.3`, IIRC this was a legitimate regression on our part. Sorry for the troubles here!

**Comment by bvaughn** [comment]
No problem! That's great to hear. Thanks Max!
